swarm/api/uri_test.go

```diff
@@ -172,5 +189,14 @@ func TestParseURI(t *testing.T) {
 		if actual.DeprecatedImmutable() != x.expectDeprecatedImmutable {
 			t.Fatalf("expected %s deprecated immutable to be %t, got %t", x.uri, x.expectDeprecatedImmutable, actual.DeprecatedImmutable())
 		}
+		if x.expectValidKey {
+			if actual.Key() == nil {
+				t.Fatalf("expected %s to return a valid key, got nil", x.uri)
+			} else {
+				if !reflect.DeepEqual(x.expectKey, actual.Key()) {
```

Comment:
Minor thing, but no need to use `reflect.DeepEqual`, `storage.Key` is just `[]byte`, you can just use `bytes.Equal`